lib/gears/surface.lua

```diff
@@ -205,6 +206,55 @@ function surface.apply_shape_bounding(draw, shape, ...)
   draw.shape_bounding = img._native
 end
 
+local function run_in_hierarchy(self, cr, width, height)
+    local function main_layout(h)
+        h:update({dpi=96}, self, width, height)
```

Comment:
Because no widget can emit any signals that this hierarchy should react to, so this code is never actually executed. ;-)
As I said, for the code you have right now, you could just use no-op-functions for these callbacks.